source/ptreferential/tests/ptref_test.cpp

```diff
@@ -301,6 +303,59 @@ BOOST_AUTO_TEST_CASE(get_indexes_test){
     BOOST_CHECK_EQUAL(indexes[1], 1);
 }
 
+BOOST_AUTO_TEST_CASE(get_impact_indexes_of_line){
+    ed::builder b("201303011T1739");
+    b.vj("A", "000001", "", true, "vj:A-1")("stop1", "08:00"_t)("stop2", "09:00"_t);
+    b.generate_dummy_basis();
+    b.finish();
+    b.data->pt_data->index();
+    b.data->pt_data->build_uri();
+
+    using btp = boost::posix_time::time_period;
+    const auto& disrup_1 = b.impact(nt::RTLevel::RealTime, "Disruption 1")
+                     .severity(nt::disruption::Effect::NO_SERVICE)
+                     .on(nt::Type_e::Line, "A")
+                     .application_periods(btp("20150928T000000"_dt, "20150928T240000"_dt))
+                     .get_disruption();
+
+    Filter filter;
+    filter.navitia_type = Type_e::Line;
+    filter.attribute = "uri";
+    filter.op = EQ;
+    filter.value = "A";
+
+    navitia::apply_disruption(disrup_1, *b.data->pt_data, *b.data->meta);
+    auto indexes = get_indexes<navitia::type::Line>(filter, Type_e::Impact, *(b.data));
+    BOOST_REQUIRE_EQUAL(indexes.size(), 1);
+    BOOST_CHECK_EQUAL(indexes[0], 0);
+
+    navitia::delete_disruption("Disruption 1", *b.data->pt_data, *b.data->meta);
+    indexes = get_indexes<navitia::type::Line>(filter, Type_e::Impact, *(b.data));
+    BOOST_REQUIRE_EQUAL(indexes.size(), 0);
+
+    const auto& disrup_2 = b.impact(nt::RTLevel::RealTime, "Disruption 2")
+                     .severity(nt::disruption::Effect::NO_SERVICE)
+                     .on(nt::Type_e::Line, "A")
+                     .application_periods(btp("20150928T000000"_dt, "20150928T240000"_dt))
+                     .get_disruption();
+
+    navitia::apply_disruption(disrup_2, *b.data->pt_data, *b.data->meta);
+    indexes = get_indexes<navitia::type::Line>(filter, Type_e::Impact, *(b.data));
+    BOOST_REQUIRE_EQUAL(indexes.size(), 1);
+    BOOST_CHECK_EQUAL(indexes[0], 0);
+
+    const auto& disrup_3 = b.impact(nt::RTLevel::RealTime, "Disruption 3")
+                     .severity(nt::disruption::Effect::NO_SERVICE)
+                     .on(nt::Type_e::Line, "A")
+                     .application_periods(btp("20150928T000000"_dt, "20150928T240000"_dt))
+                     .get_disruption();
+
+    navitia::apply_disruption(disrup_3, *b.data->pt_data, *b.data->meta);
+    indexes = get_indexes<navitia::type::Line>(filter, Type_e::Impact, *(b.data));
+    BOOST_REQUIRE_EQUAL(indexes.size(), 2);
```

Comment:
don't you want to check [1] too ? (that's why the BOOST_CHECK_COLLECTION is better I think)